### PR TITLE
Update `nf-core modules list` to show module versions

### PR DESCRIPTION
**Pull request by ErikDanielsson**
Updated `nf-core modules list` to show 
- the commit SHA
- the commit message
- the commit date

for the locally installed modules, if the `modules.json` file is present. If the `modules.json` file is missing the old version of the `list` command is shown, i.e. only the names of the installed modules

## PR checklist

 - [x] This comment contains a description of changes (with reason)
 - [x] `CHANGELOG.md` is updated
 - [x] If you've fixed a bug or added code that should be tested, add tests!
 - [ ] Documentation in `docs` is updated

### CODE REVIEW

File: nf_core/modules/module_utils.py [hunk 1]
```diff
@@ -51,6 +52,37 @@ def get_module_git_log(module_name, per_page=30, page_nbr=1, since="2020-11-25T0
...
+
+        # Parse the date returned from the API
+        date_obj = datetime.datetime.strptime(raw_date, "%Y-%m-%dT%H:%M:%SZ")
+        date = str(date_obj.date())
+
+        return message, date
+    elif response.status_code == 404:
+        raise LookupError(f"Commit '{commit_sha}' not found in 'nf-core/modules/'\n{api_url}")
+    else:
+        raise LookupError(f"Unable to fetch meta-data for commit SHA {commit_sha}")
```
Comment (KevinMenden) [inline nf_core/modules/module_utils.py:83]:
```suggestion
        raise LookupError(f"Unable to fetch metadata for commit SHA {commit_sha}")
```

**Review by KevinMenden — APPROVED** [review]
Nice, nothing to complain about! 👍

**Comment by KevinMenden** [comment]
I think the only test we have for `list` is not used on a directory. So it should be quite straight forward to add a test that runs `nf-core modules list` on a pipeline repo, which we have in the `test_modules.py` setUp function anyway. That would be nice!

**Comment by ErikDanielsson** [comment]
Thanks for the review @KevinMenden! The test sounds good, I'll add that.

**Pull request merged**